common/kvstore/KeyValueStore.h

```diff
@@ -39,6 +39,9 @@ class KeyValueStore final {
     KeyValueStore(std::string version, std::string path, std::string flavor);
     ~KeyValueStore() noexcept(false);
 
+    // Used in tests to assert that OwnedKeyValueStore cleans up reader transactions.
+    std::string getReaderLockTable();
```

Comment:
```suggestion
    std::string getReaderLockTable_forTestsOnly();
```

Reply:
Can this me the method that does the `enforce`?
E.g. `void enforceNoOutStandingReaders()`?

Reply:
I think that'd be a better API to expose

Reply:
Sure, that's fine with me.

Reply:
Done! I also moved the bulk of the functionality into a non-exported function called by the ENFORCE, so maybe it can get optimized out in release builds.